src/shared/hooks.ts

```diff
@@ -0,0 +1,44 @@
+import { RefObject, useEffect } from 'react';
+
+const EVENTS = ['mousedown', 'touchstart'];
+
+type TonClickAwayCallback = (event: Event) => void;
+
+type TuseClickAwayReferences = Array<RefObject<HTMLElement | null>>;
+
+export const useClickAway = (
+    references: TuseClickAwayReferences,
+    onClickAway?: TonClickAwayCallback
+): void => {
+    useEffect(() => {
+        const handler = (event: Event): void => {
+            if (onClickAway) {
+                const isClickAway = references.every((reference) => {
+                    const { current: element } = reference;
+
+                    return element && !element.contains(event.target as Node);
+                });
+
+                if (isClickAway) {
+                    onClickAway(event);
+                }
+            }
+        };
+
+        if (onClickAway) {
```

Comment:
If the click away will always aim to close the popover do we need to add extra logic for events and pass it in the story or can we handle this at component level? IMO its ok because the logic is quite minimal to keep that in the component state.

Reply:
This has potentially more use-cases... think of a modal for instance... that will unlikely be a popover component, so it is ok to separate the hook from the component